### PR TITLE
Check the cost of the center point of the robot and return if it's non-traversable space

### PR DESCRIPTION
**Pull request by corot**
[AB#18500](https://dev.azure.com/rapyuta-robotics/flappter/_workitems/edit/18500)

when there's a small obstacle right in front of the robot and `obstacle_approach_distance` is significant, the enlarged footprint can engulf the obstacle and so the robot is not considered in collision:

https://github.com/user-attachments/assets/e29ac988-edad-4a6d-89b2-e2b15bc093b1

This cannot happen if we check also the cell at the center of the robot, as it will enter inflated space as soon as we start moving:

https://github.com/user-attachments/assets/c962dff3-1c8d-4afe-b200-2d5d038a66d7

Other algorithms check the fp cost the same way, e.g. [Hybrid A*](https://github.com/rapyuta-robotics/smac_planner/blob/db355171656e64de55f45554286733dd5deebd02/src/collision_checker.cpp#L132).
In theory this should also provide a small increase in performance, as checking a cell is much faster than the checking the footprint lines

### CODE REVIEW

File: base_local_planner/src/costmap_model.cpp [hunk 1]
```diff
@@ -59,13 +59,15 @@ namespace base_local_planner {
     if(!costmap_.worldToMap(position.x, position.y, cell_x, cell_y))
```
Comment (plsang) [inline base_local_planner/src/costmap_model.cpp:59]: Q: is (cell_x, cell_y) the robot center?

**Review by plsang — APPROVED** [review]
the change is reasonable to me

File: base_local_planner/src/costmap_model.cpp (hunk 1 above)
Comment (corot) [reply base_local_planner/src/costmap_model.cpp:59]: yup

**Review by renan028 — APPROVED** [review]

File: base_local_planner/src/costmap_model.cpp [hunk 2]
```diff
@@ -59,13 +59,15 @@ namespace base_local_planner {
...
       return -3.0;
 
+    //check its cost, so we can skip footprint check if it's already non-traversable space
+    unsigned char cost = costmap_.getCost(cell_x, cell_y);
+    if(cost == NO_INFORMATION)
+      return -2.0;
```
Comment (renan028) [inline base_local_planner/src/costmap_model.cpp:65]: Q: should we check for lethal if center is `NO_INFORMATION`?
I think `NO_INFORMATION` may be ignored for some controllers.
For instance, in PF we have a flag to ignore it, and I think that is what we are doing.
Besides the misguiding value of the cost (255 and 254), I think `LETHAL` is a hard constraint, and `NO_INFORMATION` is a soft one.

Comment (corot) [reply base_local_planner/src/costmap_model.cpp:65]: Could be but anyway this doesn't change the logic we use now, namely return the first negative cost you find, without caring about precedence  (that is, if we find a NO_INFORMATION when checking the footprint, we return it, without searching for a LETHAL)

Comment (renan028) [reply base_local_planner/src/costmap_model.cpp:65]: true!

**Pull request merged**